### PR TITLE
Add DebugBuilder.md

### PR DESCRIPTION
**Pull request by inamiy**
This PR is a documentation for Debug Builder https://github.com/Babylonpartners/babylon-ios/pull/7645 .
(Confirmed that this doesn't need a proposal)

The main purpose of this feature is to quickly open the deeply-nested screen for faster UI debugging iteration.

#### How to make a debug-builder

1. Add a new case in `enum DebugBuilderPattern`
2. Create a (mock) builder in `DebugAuthenticatedRootBuilder`
3. Update `Settings.bundle/Root.plist` by adding a new case key
4. Set `Settings.bundle`'s userDefaults flag via `Settings.app`
5. Launch the app

### CODE REVIEW

**Review by Ben-Henshall — APPROVED** [review]

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (AliSoftware) [inline Cookbook/Technical-Documents/DebugBuilder.md]: Need to also update `Babylon/Brand/babylon/Settings.bundle/Root.plist`

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (AliSoftware) [inline Cookbook/Technical-Documents/DebugBuilder.md]:
```suggestion
4. Build and launch your app at least once after those changes, so that the Settings app can now discover the new values in the `Root.plist`
4. Set `Settings.bundle`'s userDefaults flag via `Settings.app`
```

(Note that you could also use `-debugBuilder YourCaseName` in your launch parameters in your Scheme to override those User Defaults at launch time. But might not be a good idea to suggest that since people risk forgetting to remove those and they would end up being commited)

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (AliSoftware) [inline Cookbook/Technical-Documents/DebugBuilder.md]: Update to the new name chosen once https://github.com/Babylonpartners/babylon-ios/pull/7645/files#r293449642 has been adressed

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (AliSoftware) [inline Cookbook/Technical-Documents/DebugBuilder.md]:
```suggestion
`Settings.bundle/Root.plist` AND `Babylon/Brand/babylon/Settings.bundle/Root.plist`
```

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (inamiy) [reply Cookbook/Technical-Documents/DebugBuilder.md]:
```suggestion
     case disabled
```

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (inamiy) [reply Cookbook/Technical-Documents/DebugBuilder.md]: Updated in 2a1b5da .

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (inamiy) [reply Cookbook/Technical-Documents/DebugBuilder.md]: Writing a specific path is easy to get changed, so I decided to add a note comment instead in 2a1b5da .

**Review by AliSoftware — APPROVED** [review]

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (diegopetrucci) [inline Cookbook/Technical-Documents/DebugBuilder.md]:
```suggestion
The above PR introduces `DebugAuthenticatedRootBuilder` which allows to directly show a target view builder after authentication.
```

File: Cookbook/Technical-Documents/DebugBuilder.md
Comment (diegopetrucci) [inline Cookbook/Technical-Documents/DebugBuilder.md]:
```suggestion
This feature improves the overall iteration speed for UI debugging.
```

**Review by diegopetrucci — APPROVED** [review]

**Comment by inamiy** [comment]
Babylonpartners/babylon-ios#7645 has been approved, so I will merge this PR after it's merged first.

**Comment by inamiy** [comment]
🚀

**Pull request merged**
